packages/ndla-ui/src/MyNdla/Resource/ResourceElement.tsx

```diff
@@ -324,21 +357,21 @@ const MoreIcon = styled(IconButton)<{ layout: LayoutProps }>`
   }
 `;
 
-type ResourceImageProps = {
+interface ResourceImageProps {
   alt: string;
   src: string;
-};
-interface ResourceElementProps {
+}
+export interface ResourceElementProps {
   link: string;
   title: string;
   resourceImage: ResourceImageProps;
   topics: string[];
   tags?: string[];
   description?: string;
   children?: ReactElement;
-  layout: LayoutProps;
+  layout: LayoutType;
 }
-type LayoutProps = 'list' | 'listLarger' | 'block';
+type LayoutType = 'list' | 'listLarger' | 'block' | 'dialogue';
 
 const ResourceElement = ({ link, title, tags, resourceImage, topics, layout, description }: ResourceElementProps) => {
```

Comment:
Denne filen har blitt lang nå, mest fordi den skal støtte fire forskjellige stiler og dermed mye CSS. Her er et par forslag for å redusere kodemengden:
1. Splitte ut deler av ResourceElement i nye komponenter. Tenker da på ResourceTopic og tags spesielt, men tittel og innhold kan også splittes ut.
2. Om det fortsatt er uoversiktlig kan man vurdere å lage en komponent for hver layout-type (list og listLarger er kanskje naturlig at er samme komponent fortsatt) Da vil hovedforskjellen forskjellen på komponentene hovedsakelig være plassering av elementer og styling av layout siden man kan gjenbruke komponentene fra punkt 1.